### PR TITLE
docs: add documentation about update policy annotation

### PR DESCRIPTION
**Pull request by jLopezbarb**
Signed-off-by: Javier López Barba <javier@okteto.com>

Adds documentation about the update policy annotation on compose

### CODE REVIEW

File: src/content/reference/docker-compose.mdx
Comment (pchico83) [inline src/content/reference/docker-compose.mdx:580]: @jLopezbarb were there an environment variable to define this behavior too?

Also, I am wondering if this would be a better fit for the `x-okteto-update` notation

Comment (rberrelleza) [reply src/content/reference/docker-compose.mdx:580]: Agree with @pchico83 , this should follow the x-okteto since that's what've been using for others. I'd go with `x-okteto-update-policy`

File: src/content/reference/docker-compose.mdx
Comment (rberrelleza) [inline src/content/reference/docker-compose.mdx:572]: what do we mean by 0 downtime? can we guarantee that?
```suggestion
- rolling: Will deploy a service with zero downtime. Only allowed for deployments and stateful sets.
```

Comment (rberrelleza) [reply src/content/reference/docker-compose.mdx:572]: what happens if I set this for something that's not a deployment or a stateful set?

File: src/content/reference/docker-compose.mdx
Comment (rberrelleza) [inline src/content/reference/docker-compose.mdx:574]:
```suggestion
- on-delete: Will wait until the current pod is deleted to start creating the new one. Only allowed on stateful sets.
```

File: src/content/reference/docker-compose.mdx
Comment (rberrelleza) [inline src/content/reference/docker-compose.mdx:573]:
```suggestion
- recreate: Will wait until the current pod is deleted to start creating the new one. Only allowed on deployments.
```

**Comment by pchico83** [comment]
@jLopezbarb could you update this PR to address the pending comments?

File: src/content/reference/docker-compose.mdx
Comment (jLopezbarb) [reply src/content/reference/docker-compose.mdx:572]: > what happens if I set this for something that's not a deployment or a stateful set?

It will work the same as it was working previously. This change is only for sfs and deployments

**Comment by rberrelleza** [comment]
@jLopezbarb is this ready to be merged?

**Review by rberrelleza — APPROVED** [review]

**Comment by jLopezbarb** [comment]
Yes, I was waiting for an approval

**Pull request merged**
